### PR TITLE
Add tags feature

### PR DESCRIPTION
**Pull request by matochichap**

### CODE REVIEW

**Comment by matochichap** [comment]
Tags only update on initialisation. Still need to find a way to fix this.

**Comment by jonasongg** [comment]
just to clarify, the current implementation of tags is just a panel that shows all tags that are tagged to all members currently. should we go with this or like you suggested previously that we should only allow users to tag members with a tag that already exists, but then allow them to add and delete tags also. just thought of a problem with this approach: if a user wants to create a brand new member with a new tag, how should we handle this?

also, might want to consider creating a uniquetaglist class just like uniquepersonlist to handle the getTagList
image
so would have a getTagList method also in the picture above. but might be unnecessarily complicated also, up to your judgement!

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -165,6 +169,24 @@ public ObservableList<Applicant> getApplicantList() {
         return applicants.asUnmodifiableObservableList();
     }
 
+    @Override
```
Comment (jonasongg) [inline src/main/java/seedu/address/model/AddressBook.java:172]: This makes sense to me, good SLAP in abstracting out the `updateTags` method

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 2]
```diff
@@ -92,6 +92,11 @@ public void getFilteredApplicantList_modifyList_throwsUnsupportedOperationExcept
         assertThrows(UnsupportedOperationException.class, () -> logic.getFilteredApplicantList().remove(0));
     }
 
+    @Test
```
Comment (jonasongg) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:95]: Nice that you added test cases for this

**Review by jonasongg — APPROVED** [review]
Looks good to me! Nice that you didn't forget about adding test cases :)

**Pull request merged**